gibbon-compiler/src/Gibbon/Passes/SolveLayoutConstrs.hs

```diff
@@ -55,87 +55,507 @@ solveConstrs constrs = do
       "Running docplex program in file " ++
       fp ++ " exited with error code  " ++ show n
 
-pythonCodegen :: [Constr] -> PassM String
-pythonCodegen constrs
-  -- There should be a map mapping indexes to the generated variables
-  -- 3 maps for different constraints
-  -- Here have a map to seperate the soft constraints and pass as usual
- = do
-  let idxs =
-        L.nub $
-        P.concatMap
-          (\a ->
-             case a of
-               Soft ((a, b), _) -> [a, b]
-               _                -> [])
-          constrs
-  let softConstr =
-        P.concatMap
-          (\a ->
-             case a of
-               Soft a -> [a]
-               _      -> [])
-          constrs
-  let immediateConstrVariables =
-        L.nub $
-        P.concatMap
-          (\a ->
-             case a of
-               Imm (a, b) -> [a, b]
-               _          -> [])
-          constrs
+-- pythonCodegen :: [Constr] -> PassM String
```

Comment:
Large chink of this file is comments. Let's remove them? (whenever you have time to get back to this; I hope this PR will be cleaned up and merged...)